### ISSUE TITLE
Flaky test: MvcTemplate_NoAuth_Works_NetFramework_ForFSharpTemplate

### ISSUE THREAD

**Issue by ryanbrandenburg** [aspnet/Templating#645]
This test [fails](http://aspnetci/viewLog.html?buildId=517743&buildTypeId=Releases_22xPublic_Win2012) occasionally with the following error:
```
System.Exception : Process exited with code 1
StdErr: 
StdOut: Microsoft (R) Build Engine version 15.7.179.6572 for .NET Core
Copyright (C) Microsoft Corporation. All rights reserved.
C:\b\w\33bdfc1cae7b2a38\.dotnet\x64\sdk\2.1.302\Sdks\Microsoft.NET.Sdk\targets\Microsoft.PackageDependencyResolution.targets(198,5): error : Assets file 'C:\b\w\33bdfc1cae7b2a38\modules\Templating\test\Templates.Test\bin\Release\netcoreapp2.2\TestTemplates\AspNet.Template.4d94c49c169c48f1935fa5cb592373df\obj\project.assets.json' not found. Run a NuGet package restore to generate this file. [C:\b\w\33bdfc1cae7b2a38\modules\Templating\test\Templates.Test\bin\Release\netcoreapp2.2\TestTemplates\AspNet.Template.4d94c49c169c48f1935fa5cb592373df\AspNet.Template.4d94c49c169c48f1935fa5cb592373df.fsproj]
Build FAILED.
C:\b\w\33bdfc1cae7b2a38\.dotnet\x64\sdk\2.1.302\Sdks\Microsoft.NET.Sdk\targets\Microsoft.PackageDependencyResolution.targets(198,5): error : Assets file 'C:\b\w\33bdfc1cae7b2a38\modules\Templating\test\Templates.Test\bin\Release\netcoreapp2.2\TestTemplates\AspNet.Template.4d94c49c169c48f1935fa5cb592373df\obj\project.assets.json' not found. Run a NuGet package restore to generate this file. [C:\b\w\33bdfc1cae7b2a38\modules\Templating\test\Templates.Test\bin\Release\netcoreapp2.2\TestTemplates\AspNet.Template.4d94c49c169c48f1935fa5cb592373df\AspNet.Template.4d94c49c169c48f1935fa5cb592373df.fsproj]
    0 Warning(s)
    1 Error(s)
Time Elapsed 00:00:01.10
   at Templates.Test.Helpers.ProcessEx.WaitForExit(Boolean assertSuccess) in /_/test/Templates.Test/Helpers/ProcessEx.cs:line 150
   at Templates.Test.Helpers.AspNetProcess..ctor(ITestOutputHelper output, String workingDirectory, String projectName, String targetFrameworkOverride, Boolean publish) in /_/test/Templates.Test/Helpers/AspNetProcess.cs:line 66
   at Templates.Test.TemplateTestBase.StartAspNetProcess(String targetFrameworkOverride, Boolean publish) in /_/test/Templates.Test/Helpers/TemplateTestBase.cs:line 205
   at Templates.Test.MvcTemplateTest.MvcTemplate_NoAuthImpl(String targetFrameworkOverride, String languageOverride, Boolean noHttps) in /_/test/Templates.Test/MvcTemplateTest.cs:line 73
------- Stdout: -------
==> C:\b\w\33bdfc1cae7b2a38\.dotnet\x64\dotnet.exe new mvc --debug:custom-hive "C:\b\w\33bdfc1cae7b2a38\modules\Templating\test\Templates.Test\bin\Release\netcoreapp2.2\.templateengine" --target-framework-override net461 -lang F# [C:\b\w\33bdfc1cae7b2a38\modules\Templating\test\Templates.Test\bin\Release\netcoreapp2.2\TestTemplates\AspNet.Template.4d94c49c169c48f1935fa5cb592373df]
The template "ASP.NET Core Starter Web App (Model-View-Controller)" was created successfully.
This template contains technologies from parties other than Microsoft, see https://aka.ms/aspnetcore-template-3pn-210 for details.
Processing post-creation actions...
Running 'dotnet restore' on C:\b\w\33bdfc1cae7b2a38\modules\Templating\test\Templates.Test\bin\Release\netcoreapp2.2\TestTemplates\AspNet.Template.4d94c49c169c48f1935fa5cb592373df\AspNet.Template.4d94c49c169c48f1935fa5cb592373df.fsproj...
  Restoring packages for C:\b\w\33bdfc1cae7b2a38\modules\Templating\test\Templates.Test\bin\Release\netcoreapp2.2\TestTemplates\AspNet.Template.4d94c49c169c48f1935fa5cb592373df\AspNet.Template.4d94c49c169c48f1935fa5cb592373df.fsproj...
C:\b\w\33bdfc1cae7b2a38\.dotnet\x64\sdk\2.1.302\NuGet.targets(114,5): error : Unable to load the service index for source https://dotnetfeed.blob.core.windows.net/dotnet-core/index.json. [C:\b\w\33bdfc1cae7b2a38\modules\Templating\test\Templates.Test\bin\Release\netcoreapp2.2\TestTemplates\AspNet.Template.4d94c49c169c48f1935fa5cb592373df\AspNet.Template.4d94c49c169c48f1935fa5cb592373df.fsproj]
C:\b\w\33bdfc1cae7b2a38\.dotnet\x64\sdk\2.1.302\NuGet.targets(114,5): error :   No such host is known [C:\b\w\33bdfc1cae7b2a38\modules\Templating\test\Templates.Test\bin\Release\netcoreapp2.2\TestTemplates\AspNet.Template.4d94c49c169c48f1935fa5cb592373df\AspNet.Template.4d94c49c169c48f1935fa5cb592373df.fsproj]
C:\b\w\33bdfc1cae7b2a38\.dotnet\x64\sdk\2.1.302\NuGet.targets(114,5): error : NuGet.Protocol.Core.Types.FatalProtocolException: Unable to load the service index for source https://dotnetfeed.blob.core.windows.net/dotnet-core/index.json. ---> System.Net.Http.HttpRequestException: No such host is known ---> System.Net.Sockets.SocketException: No such host is known [C:\b\w\33bdfc1cae7b2a38\modules\Templating\test\Templates.Test\bin\Release\netcoreapp2.2\TestTemplates\AspNet.Template.4d94c49c169c48f1935fa5cb592373df\AspNet.Template.4d94c49c169c48f1935fa5cb592373df.fsproj]
C:\b\w\33bdfc1cae7b2a38\.dotnet\x64\sdk\2.1.302\NuGet.targets(114,5): error :    at System.Net.Http.ConnectHelper.ConnectAsync(String host, Int32 port, CancellationToken cancellationToken) [C:\b\w\33bdfc1cae7b2a38\modules\Templating\test\Templates.Test\bin\Release\netcoreapp2.2\TestTemplates\AspNet.Template.4d94c49c169c48f1935fa5cb592373df\AspNet.Template.4d94c49c169c48f1935fa5cb592373df.fsproj]
C:\b\w\33bdfc1cae7b2a38\.dotnet\x64\sdk\2.1.302\NuGet.targets(114,5): error :    --- End of inner exception stack trace --- [C:\b\w\33bdfc1cae7b2a38\modules\Templating\test\Templates.Test\bin\Release\netcoreapp2.2\TestTemplates\AspNet.Template.4d94c49c169c48f1935fa5cb592373df\AspNet.Template.4d94c49c169c48f1935fa5cb592373df.fsproj]
C:\b\w\33bdfc1cae7b2a38\.dotnet\x64\sdk\2.1.302\NuGet.targets(114,5): error :    at System.Net.Http.ConnectHelper.ConnectAsync(String host, Int32 port, CancellationToken cancellationToken) [C:\b\w\33bdfc1cae7b2a38\modules\Templating\test\Templates.Test\bin\Release\netcoreapp2.2\TestTemplates\AspNet.Template.4d94c49c169c48f1935fa5cb592373df\AspNet.Template.4d94c49c169c48f1935fa5cb592373df.fsproj]
C:\b\w\33bdfc1cae7b2a38\.dotnet\x64\sdk\2.1.302\NuGet.targets(114,5): error :    at System.Threading.Tasks.ValueTask`1.get_Result() [C:\b\w\33bdfc1cae7b2a38\modules\Templating\test\Templates.Test\bin\Release\netcoreapp2.2\TestTemplates\As
...
```
Other tests within that build may have failed with a similar message, but they are not listed here. Check the link above for more info.

CC @Eilon,@muratg,@mkArtakMSFT

This issue was made automatically. If there is a problem contact @ryanbrandenburg.

**Comment by ryanbrandenburg** [comment]
MvcTemplate_NoAuth_Works_NetFramework_ForFSharpTemplate [failed again](http://aspnetci/viewLog.html?buildId=517599&buildTypeId=Lite_UniverseTest).

This comment was made automatically. If there is a problem contact @ryanbrandenburg.

**Comment by ryanbrandenburg** [comment]
MvcTemplate_NoAuth_Works_NetFramework_ForFSharpTemplate [failed again](http://aspnetci/viewLog.html?buildId=518163&buildTypeId=Lite_UniverseTest).

This comment was made automatically. If there is a problem contact @ryanbrandenburg.

**Comment by ryanbrandenburg** [comment]
MvcTemplate_NoAuth_Works_NetFramework_ForFSharpTemplate [failed again](http://aspnetci/viewLog.html?buildId=518450&buildTypeId=Lite_UniverseTest).

This comment was made automatically. If there is a problem contact @ryanbrandenburg.

**Comment by ryanbrandenburg** [comment]
MvcTemplate_NoAuth_Works_NetFramework_ForFSharpTemplate [failed again](http://aspnetci/viewLog.html?buildId=518926&buildTypeId=Lite_UniverseTest).

This comment was made automatically. If there is a problem contact @ryanbrandenburg.

**Comment by ryanbrandenburg** [comment]
MvcTemplate_NoAuth_Works_NetFramework_ForFSharpTemplate [failed again](http://aspnetci/viewLog.html?buildId=519091&buildTypeId=Lite_UniverseTest).

This comment was made automatically. If there is a problem contact @ryanbrandenburg.

**Comment by ryanbrandenburg** [comment]
MvcTemplate_NoAuth_Works_NetFramework_ForFSharpTemplate [failed again](http://aspnetci/viewLog.html?buildId=519339&buildTypeId=Lite_UniverseTest).

This comment was made automatically. If there is a problem contact @ryanbrandenburg.

**Comment by ryanbrandenburg** [comment]
MvcTemplate_NoAuth_Works_NetFramework_ForFSharpTemplate [failed again](http://aspnetci/viewLog.html?buildId=519492&buildTypeId=Lite_UniverseTest).

This comment was made automatically. If there is a problem contact @ryanbrandenburg.

**Comment by ryanbrandenburg** [comment]
MvcTemplate_NoAuth_Works_NetFramework_ForFSharpTemplate [failed again](http://aspnetci/viewLog.html?buildId=519572&buildTypeId=Lite_UniverseTest).

This comment was made automatically. If there is a problem contact @ryanbrandenburg.

**Comment by natemcmaster** [comment]
It looks like the reason this has changed is different from the original post.
```
System.Exception : Process exited with code 2097158
StdErr: Some partially matched templates may not support these input switches:
  --target-framework-override
  net461
```

**Comment by ryanbrandenburg** [comment]
MvcTemplate_NoAuth_Works_NetFramework_ForFSharpTemplate [failed with about the same error](http://aspnetci/viewLog.html?buildId=520895&buildTypeId=Lite_UniverseTest).

This comment was made automatically. If there is a problem contact @ryanbrandenburg.

**Comment by ryanbrandenburg** [comment]
MvcTemplate_NoAuth_Works_NetFramework_ForFSharpTemplate [failed with about the same error](http://aspnetci/viewLog.html?buildId=522188&buildTypeId=Lite_UniverseTest).

This comment was made automatically. If there is a problem contact @ryanbrandenburg.

**Comment by ryanbrandenburg** [comment]
@ajaybhargavb, he's a templating flaky test that we don't know the cause of yet.

**Comment by ryanbrandenburg** [comment]
MvcTemplate_NoAuth_Works_NetFramework_ForFSharpTemplate [failed with about the same error](http://aspnetci/viewLog.html?buildId=523413&buildTypeId=Lite_UniverseTest).

This comment was made automatically. If there is a problem contact @ryanbrandenburg.

**Comment by ryanbrandenburg** [comment]
MvcTemplate_NoAuth_Works_NetFramework_ForFSharpTemplate [failed with about the same error](http://aspnetci/viewLog.html?buildId=524711&buildTypeId=Lite_UniverseTest).

This comment was made automatically. If there is a problem contact @ryanbrandenburg.

**Comment by ryanbrandenburg** [comment]
MvcTemplate_NoAuth_Works_NetFramework_ForFSharpTemplate [failed with about the same error](http://aspnetci/viewLog.html?buildId=524676&buildTypeId=Lite_UniverseTest).

This comment was made automatically. If there is a problem contact @ryanbrandenburg.

**Comment by ryanbrandenburg** [comment]
Judging by the frequency with which this failed before I think this was fixed by my recent blast of fixes to Templating. I'm not exactly sure what I did to fix it though. @ajaybhargavb I'll leave it to you to decide if you want to dig more or call it fixed.

**Comment by ajaybhargavb** [comment]
Fixed by @ryanbrandenburg with d95f9716932cae55933c0a4700fec3513c55c92f
